test/cmd/ephemeral/ephemeral.sh

```diff
@@ -0,0 +1,547 @@
+
+    run_app $1
+    os::cmd::try_until_text 'oc logs dc/bob' 'Found incomplete cluster'
+    os::cmd::expect_success 'oc create -f "$file"'
+    rm $file
+
+    os::cmd::try_until_text 'oc logs dc/bob' "Found cluster"
+    cleanup_app
+    cleanup_cluster
+}
+
+function scaled_app_completed_cluster_remains() {
+    echo running scaled_app_completed_cluster_remains
+    set_defaults
+    clear_spark_sleep
+    run_app "bob"
+
+    os::cmd::try_until_success 'oc get dc "$MASTER_DC"' $((2*minute))
+    os::cmd::try_until_success 'oc get dc "$WORKER_DC"'
+
+    DRIVER=$(oc get pod -l deploymentconfig=bob --template='{{index .items 0 "metadata" "name"}}')
+    os::cmd::try_until_text 'oc logs "$DRIVER"' 'Running Spark' $((5*minute))
+    os::cmd::expect_success 'oc scale dc/bob --replicas=2'
+    	
+    os::cmd::try_until_text 'oc logs "$DRIVER"' 'Deleting cluster' $((5*minute))
+    os::cmd::try_until_text 'oc logs "$DRIVER"' 'driver replica count > 0'
+    os::cmd::try_until_text 'oc logs "$DRIVER"' 'cluster not deleted'
+
+    cleanup_app wait_for_cluster_del
+}
+
+function redeploy_cluster_removed() {
+    echo running redeploy_cluster_removed
+    set_defaults
+    set_app_exit
+    set_test_mode 
+    run_app "bob"
+
+    os::cmd::try_until_success 'oc get dc "$MASTER_DC"' $((2*minute))
+    os::cmd::try_until_success 'oc get dc "$WORKER_DC"'
+
+    DRIVER=$(oc get pod -l deploymentconfig=bob --template='{{index .items 0 "metadata" "name"}}')
+    os::cmd::try_until_failure 'oc get pod bob-1-deploy'
+    os::cmd::expect_success 'oc deploy dc/bob --latest'
+
+    os::cmd::try_until_text 'oc logs "$DRIVER"' "Deleting cluster" $((5*minute))
+    os::cmd::try_until_failure 'oc get pod "$DRIVER"'
+
+    DRIVER=$(oc get pod -l deployment=bob-2 --template='{{index .items 0 "metadata" "name"}}')
+    os::cmd::try_until_text 'oc logs "$DRIVER"' "Didn't find cluster"
+    os::cmd::try_until_text 'oc logs "$DRIVER"' "Waiting for spark master"
+
+    cleanup_app wait_for_cluster_del
+}
+
+source "${SCRIPT_DIR}/../../../hack/lib/init.sh"
+trap os::test::junit::reconcile_output EXIT
+
+os::test::junit::declare_suite_start "cmd/cluster"
+os::cmd::expect_success 'oc new-project "$PROJECT"'
+os::cmd::expect_success 'oc create sa oshinko'
+os::cmd::expect_success 'oc policy add-role-to-user admin system:serviceaccount:"$PROJECT":oshinko -n "$PROJECT"'
+
+# The ip address of the internal registry may be passed as the first argument to
+# support running against an openshift that is not "oc cluster up". In the case of
+# "oc cluster up", the docker on the host is available from openshift so no special
+# pushes of images have to be done. In the case of a "normal" openshift cluster, the
+# image we'll use for build has to be available as an imagestream.
+if [ "$#" -eq 0 ]; then
+    os::cmd::expect_success 'oc new-build --name=play "$S2I_TEST_IMAGE" --binary'
+else
+    docker login -u tmckay -p $(oc whoami -t) $1
```

Comment:
oops, can't have tmckay here.  Needs to be an argument.  For now, if you try this, put yourself here (or add a tmckay user :) )